ffi/framebuffer_linux.lua

```diff
@@ -219,6 +241,8 @@ function framebuffer:init()
                            0)
     assert(tonumber(ffi.cast("intptr_t", self.data)) ~= C.MAP_FAILED,
            "can not mmap() framebuffer")
+
+    -- New invariant: self.bb MUST NOT be cached by anyone if we reinit, as the old one will point to junk now.
```

Comment:
I think you use "invariant" a few times before. English not being my first language, can you explicite what you mean when you use it, like in here?

Reply:
It denotes location of contract that is invariable. Ie when an invariant is introduced, it implies there's a contract too - or warns that this end might be at odds with invariant of pre-existing contract somewhere else. It's useful to mark invariant because then you just grep source for 'contract|invariant' and notice if some are at odds with each other.

Reply:
OK :) So it doesn't mean anything more than what I would have written at your place:
`-- Note: don't ever cache self.bb, as we may replace it at any time` :)
It's just using some kind of rare keyword instead of my "Nore", to allow grep'ing or indexing.

Reply:
Invariant just means not changing. This is awkward at best imho.

Reply:
Btw, we use the `@warning` annotation for that.

Reply:
I won't pretend this isn't pretentious as all hell (don't make me use something really awkward like "congruence"!), but indeed this is about caveats that specifically deal with "spooky effect at a distance" one looks specifically for. `note` is indeed far too general.

Reply:
Like I said, our standard is:
```
-- @warning This will blow up.
```

There's no need to tag a note.